### ISSUE TITLE
[java] ImmutableField false positive with Spring @Value

### ISSUE THREAD

**Issue by mikebell90** [pmd/pmd#2156]
**Affects PMD Version:** 

6.20

Longtime PMD user, never ran into this one before

**Rule:**

ImmutableField

**Description:**

As of 6.20, our Config classes in Spring get flagged. They look like

**Code Sample demonstrating the issue:**

```
 @Value("${ot.credentials.client.secret.rotation.lognochange:#{false}}")
    private boolean logNoChange = false;
```

Admittedly in this case, the default (false) would work. This is not an option for other defaults lijke Duration. Also we could add setters, but they would just add boilerplate.

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

maven

**Comment by mikebell90** [comment]
We could also redo these config holders in the constructor, but were trying to avoid this.

**Comment by yatsenko-ihor** [comment]
@mikebell90 could you please explain why you are avoid constructor injection? 

In my opinion such approach when private variable have value which spring inject using reflection not reflect real oop approach. It's not clear, so may be not make a big sense to use some specific rule which means that pmd is encourages oop violations.

you can declare variable with default package

**Comment by adangel** [comment]
Was duplicated by #3981
